internal/mage/sdk/rust.go

```diff
@@ -227,7 +227,7 @@ func (Rust) rustBase(ctx context.Context, c *dagger.Client, image string) *dagge
 	base := c.
 		Container().
 		From(image).
-		WithMountedCache("~/.cargo", c.CacheVolume("rust-cargo")).
+		WithMountedCache("~/.cargo", c.CacheVolume("rust-cargo-"+image)).
```

Comment:
Huh, mildly surprised that using `~` works here. Wouldn't we need to explicitly expand that somewhere? From the code I don't see that happening, so I wonder if this is mounting to literally `./~/.cargo` and it's working without the cache atm.

Reply:
Oh it probably doesn't work, good catch, I'll just fix that too while I'm modifying this